api/specs/web-server/_projects_groups.py

```diff
@@ -24,6 +26,22 @@
 )
 
 
+@router.post(
+    "/projects/{project_id}:share",
+    response_model=Envelope[ProjectShareAccepted],
+    status_code=status.HTTP_202_ACCEPTED,
+    responses={
+        status.HTTP_202_ACCEPTED: {
+            "description": "The request to share the project has been accepted, but the actual sharing process has to be confirmd."
+        }
+    },
+)
+async def share_project(
+    _path: Annotated[ProjectPathParams, Depends()],
+    _body: ProjectShare,
+): ...
```

Comment:
just a bit confused by the fact that:
- project_groups is as far as I know the new name for project access rights (which is a bit confusing already)
- this share function is to share with anyone via email, so actually nothing to do with a group (maybe the primary one then)
- we do have some sharing functions already (not sure where they are now), would it not make sense to move that entrypoint there?

This is just a thought.

Reply:
@sanderegg I agree. Two things:

- one problem is that we are in a transition in which the projects domain is still fragmented: Old groups module and access_rights (which refer to the same concept) are still living together.  Merging them is part of the refactoring issue in https://github.com/ITISFoundation/osparc-simcore/issues/6595
- the sharing logic (of any sharing entity e.g. projects, services etc) will introduce "confirmation" of the sharee. 

In any case, I took action in this PR and started merging at least the controller and schemas